DLSv2/Utils/DLSExtensions.cs

```diff
@@ -121,7 +121,121 @@ internal static SirenSetting GetDefaultSirenSetting(this Vehicle veh)
+                        vehicle.SetModkitModIndex(kit.Type, kit.Index);
+
+                // Sets the yield setting
+                if (mode.Yield.Enabled) shouldYield = true;
+
+                // Sets the indicators
+                if (mode.Indicators != null)
+                {
+                    switch (mode.Indicators.ToLower())
+                    {
+                        case "off":
+                            managedVehicle.IndStatus = VehicleIndicatorLightsStatus.Off;
+                            break;
+                        case "rightonly":
+                            managedVehicle.IndStatus = VehicleIndicatorLightsStatus.RightOnly;
+                            break;
+                        case "leftonly":
+                            managedVehicle.IndStatus = VehicleIndicatorLightsStatus.LeftOnly;
+                            break;
+                        case "both":
+                            managedVehicle.IndStatus = VehicleIndicatorLightsStatus.Both;
+                            break;
+                    }
+                    managedVehicle.Vehicle.SetIndicator(managedVehicle.IndStatus);
+                }
+            }
+
+            // Adjust time multiplier if a drift is configured
+            var newTimeMultiplier = SyncManager.GetAdjustedMultiplier(vehicle, eL.TimeMultiplier);
+            if (newTimeMultiplier.HasValue) eL.TimeMultiplier = newTimeMultiplier.Value;
+
+            // Set enabled extras first, then disabled extras second, because <extraIncludes> in vehicles.meta 
+            // can cause enabling one extra to enable other linked extras. By disabling second, we turn back off 
+            // any extras that are explicitly set to be turned off.
+            foreach (var extra in extras.OrderByDescending(e => e.Value))
+            {
+                if (!vehicle.HasExtra(extra.Key)) continue;
+                if (!managedVehicle.ManagedExtras.ContainsKey(extra.Key)) managedVehicle.ManagedExtras[extra.Key] = vehicle.IsExtraEnabled(extra.Key);
+                vehicle.SetExtra(extra.Key, extra.Value);
+            }
+
+            // Reset any extras not specified by the current mode back to their previous setting before they were set by any mode
+            var extrasToReset = managedVehicle.ManagedExtras.Keys.Where(x => !extras.ContainsKey(x)).ToList();
+            foreach (var extra in extrasToReset)
+            {
+                if (vehicle.HasExtra(extra)) vehicle.SetExtra(extra, managedVehicle.ManagedExtras[extra]);
+                managedVehicle.ManagedExtras.Remove(extra);
+            }
+
+            vehicle.ShouldVehiclesYieldToThisVehicle = shouldYield;
+
+            if (!Entrypoint.ELUsedPool.ContainsKey(key))
+                Entrypoint.ELUsedPool.Add(key, eL);
+
+            managedVehicle.Vehicle.EmergencyLightingOverride = eL;
+        }
+
+        /*internal static void DebugCurrentModes(this Vehicle vehicle)
```

Comment:
Is this disabled just because things changed so the logging doesn't work currently? Seems like it will be very helpful for testing if the changes work right. Do you plan to add it back?

Reply:
Yeah, disabled due to the restructuring. Will work it back on before merging.